icepyx/core/read.py

```diff
@@ -264,19 +264,21 @@ class Read:
 
     Parameters
     ----------
-    data_source : string
-        A string with a full file path or full directory path to ICESat-2 hdf5 (.h5) format files.
-        Files within a directory must have a consistent filename pattern that includes the "ATL??" data product name.
-        Files must all be within a single directory.
+    data_source : string, List
+        A string or list which specifies the files to be read. The string can be either: 1) the path of a single file 2) the path to a directory or 3) a [glob string](https://docs.python.org/3/library/glob.html).
 
     product : string
         ICESat-2 data product ID, also known as "short name" (e.g. ATL03).
         Available data products can be found at: https://nsidc.org/data/icesat-2/data-sets
+        **Depreciation warning:** This argument is no longer required and will be depreciated in version 1.0.0. The dataset product is read from the file metadata.
 
     filename_pattern : string, default 'ATL{product:2}_{datetime:%Y%m%d%H%M%S}_{rgt:4}{cycle:2}{orbitsegment:2}_{version:3}_{revision:2}.h5'
         String that shows the filename pattern as required for Intake's path_as_pattern argument.
```

Comment:
```suggestion
    filename_pattern : string, default None
        String that shows the filename pattern as previously required for Intake's path_as_pattern argument.
```
Note: Intake removed in #438